### ISSUE TITLE
RuntimeError on training GTA

### ISSUE THREAD

**Issue by taein98** [sw32-seo/GTA#3]
I appreciate your nice work on retrosynthesis. However, I get a RuntimeError message on running train.py when I tried to start from preprocessing the USPTO-50k_no_rxn_dataset. (torchtext 0.4.0, pytorch 1.8.2, python 3.8.0, cudatoolkit 11.1.7)

![image](https://user-images.githubusercontent.com/75014091/215364129-6bb63d05-c0e4-4900-9f9a-46ed648a7a3c.png)

And I also found a similar issue #1 and the different dtypes on running loss

![image](https://user-images.githubusercontent.com/75014091/215364619-49417529-50c7-4d5e-92fd-e53e00df97b2.png)
![image](https://user-images.githubusercontent.com/75014091/215364496-4ca591a1-f8c1-4fbd-9466-55485bc600d8.png)

I would deeply appreciate your effort if you help debug!

**Comment by sw32-seo** [comment]
Could you give me the full error message and the command you used?
I think the error is not coming from loss.div() but happens while backprop.

**Comment by taein98** [comment]
I agree with you. That was an error message from debugging, And here is the full error message. (below error repeats)

![image](https://user-images.githubusercontent.com/75014091/215386400-23af568a-a8e1-472d-888e-84dc594ce401.png)

And the command I used

![image](https://user-images.githubusercontent.com/75014091/216524810-3f8aff77-4b20-47de-b58e-b88b157c30ca.png)

python  train.py -data data/${dataset}/${dataset} \
                 -save_model experiments/${dataset}_${model_name} \
                 -seed 2023 -gpu_ranks 0 -world_size 1 \
                 -save_checkpoint_steps 1000  -keep_checkpoint 11 \
                 -train_steps 1000 -valid_steps 1000 -report_every 1000 \
                 -param_init 0 -param_init_glorot \
                 -batch_size 4 -batch_type tokens -normalization tokens \
                 -dropout 0.3 -max_grad_norm 0 -accum_count 4 \
                 -optim adam -adam_beta1 0.9 -adam_beta2 0.998 \
                 -decay_method noam -warmup_steps 8000  \
                 -learning_rate 2 -label_smoothing 0.0 \
                 -enc_layers 6 -dec_layers 6 -rnn_size 256 -word_vec_size 256 \
                 -encoder_type transformer -decoder_type transformer \
                 -share_embeddings -position_encoding -max_generator_batches 0 \
                 -global_attention general -global_attention_function softmax \
                 -self_attn_type scaled-dot -max_relative_positions 4 \
                 -heads 8 -transformer_ff 2048 -max_distance 1 2 3 4 \
                 -early_stopping 40 -alpha 1.0 \
                 -tensorboard -tensorboard_log_dir runs/${dataset}_${model_name} 2>&1 | tee train_$model_name.log
                 
 
The pytorch 1.4.0 is not compatible with our GPU (RTX 3090)
![image](https://user-images.githubusercontent.com/75014091/216529765-56590815-ff16-49e0-9eff-cf40707d7242.png)

And the oldest version of pytorch (1.7.0) which is compatible with cuda 11 doesn't work, too.

![image](https://user-images.githubusercontent.com/75014091/216530946-a444d93a-d834-48bb-8c29-1d9d8cf087a1.png)

**Comment by YoujunZhao** [comment]
I got the same problem when I used the code for training. Have you solve problem yet?

**Comment by Seojin-Kim** [comment]
Although I'm not the author of this paper, I'm gonna share my solution below.

You can simply replace "cross_mask" of the 279th line of onmt/utils/loss.py with "cross_mask.float()"

The main issue was the type mismatch of input (float) and target (integer) within MSE loss.

**Comment by sw32-seo** [comment]
Thank you @Seojin-Kim you helped me a lot!
